plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts

```diff
@@ -0,0 +1,313 @@
+      );
+    }
+
+    const expectedEntityFiles = this.options.files;
+
+    const { repositories } = await getOrganizationEntities(
+      client,
+      org,
+      expectedEntityFiles,
+    );
+
+    const matching = repositories.filter(
+      r =>
+        !r.isArchived &&
+        repoSearchPath.test(r.name) &&
+        r.defaultBranchRef?.name,
+    );
+
+    const duration = ((Date.now() - startTimestamp) / 1000).toFixed(1);
+
+    logger.info(
+      `Read ${repositories.length} GitHub repositories (${matching.length} matching the pattern) in ${duration} seconds`,
+    );
+
+    const files = expectedEntityFiles.map(fileName => ({
+      name: fileName,
+      key: camelCase(fileName),
+    }));
+
+    const repoEntities: Array<LocationSpec | Array<LocationSpec>> = [];
+
+    for (const entity of matching) {
+      for (const { name: fileName, key } of files) {
+        const entityObj: any = entity[key as keyof typeof entity];
+        if (entityObj) {
+          const locationEntity = this.createLocationSpec({
+            url: entity.url,
+            branch: entity.defaultBranchRef?.name || '',
+            file: fileName,
+          });
+          repoEntities.push(locationEntity);
+        }
+      }
+    }
+
+    const entityLocations = repoEntities.flat();
+
+    if (entityLocations.length === 0) {
+      logger.info(`No valid entities found by ${this.getProviderName()}.`);
+      return false;
+    }
+
+    await this.connection.applyMutation({
+      type: 'full',
+      entities: entityLocations.map(location => ({
+        locationKey: this.getProviderName(),
+        entity: locationSpecToLocationEntity({ location }),
+      })),
+    });
+
+    logger.info(
+      `Adding ${entityLocations.length} valid entities to the catalog`,
+    );
+
+    return true;
+  }
+
+  private createLocationSpec({
+    url,
+    branch,
+    file,
+  }: CreateLocationSpec): LocationSpec {
+    return {
+      type: 'url',
+      target: `${url}/blob/${branch}/${file}`,
+      presence: 'optional',
+    };
+  }
+}
+
+/*
+ * Helpers
+ */
+
+export function parseUrl(urlString: string): {
+  org: string;
+  repoSearchPath: RegExp;
+  catalogPath: string;
+  branch: string;
+  host: string;
+} {
+  const url = new URL(urlString);
+  const path = url.pathname.substr(1).split('/');
+
+  // /backstage/techdocs-*/blob/master/catalog-info.yaml
+  // can also be
+  // /backstage
+  if (path.length > 2 && path[0].length && path[1].length) {
+    return {
+      org: decodeURIComponent(path[0]),
+      repoSearchPath: escapeRegExp(decodeURIComponent(path[1])),
+      branch: decodeURIComponent(path[3]),
+      catalogPath: `/${decodeURIComponent(path.slice(4).join('/'))}`,
+      host: url.host,
+    };
+  } else if (path.length === 1 && path[0].length) {
+    return {
+      org: decodeURIComponent(path[0]),
+      host: url.host,
+      repoSearchPath: escapeRegExp('*'),
+      catalogPath: '/catalog-info.yaml',
```

Comment:
Hm now i'm not sure - how does this interact with the other `files` parameter up there?

Reply:
removed unnecessary code